File: backend/app/api/routes/predictions.py
"""Prediction routes."""
from __future__ import annotations

import uuid

from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.ext.asyncio import AsyncSession

from app.deps import CurrentUser, DBSession
from app.schemas.prediction import PredictionOut, PredictionRequest
from app.services import prediction_service

router = APIRouter(prefix="/predictions", tags=["predictions"])


@router.post("", response_model=PredictionOut, status_code=status.HTTP_201_CREATED)
async def create_prediction(
    data: PredictionRequest, db: DBSession, current: CurrentUser
) -> PredictionOut:
    try:
        prediction = await prediction_service.predict(db, data.compound_id, data.method_id)
    except ValueError as exc:
        raise HTTPException(status.HTTP_404_NOT_FOUND, str(exc)) from exc
    return PredictionOut.model_validate(prediction)

File: frontend/src/api/predictions.ts
import { apiClient } from './client';
import type { Prediction } from '@/types';

export const predictionsApi = {
  create: async (compoundId: string, methodId: string) => {
    const { data } = await apiClient.post<Prediction>('/predictions', {
      compound_id: compoundId,
      method_id: methodId,
    });
    return data;
  },
};
